pkg/clusterversion/cv.go

```diff
@@ -64,44 +68,67 @@ func (c *clusterVersionClient) GetClusterVersion() (*configv1.ClusterVersion, er
 	return cv, err
 }
 
-func (c *clusterVersionClient) EnsureDesiredVersion(uc *upgradev1alpha1.UpgradeConfig) (bool, error) {
+func (c *clusterVersionClient) EnsureDesiredConfig(uc *upgradev1alpha1.UpgradeConfig) (bool, error) {
 	clusterVersion, err := c.GetClusterVersion()
 	if err != nil {
 		return false, err
 	}
 
-	// Move the cluster to the same channel first
 	desired := uc.Spec.Desired
-	if clusterVersion.Spec.Channel != desired.Channel {
-		clusterVersion.Spec.Channel = desired.Channel
-		err = c.client.Update(context.TODO(), clusterVersion)
-		if err != nil {
-			return false, err
+
+	// Commence upgrade using version and channel
+	// Move the cluster to the same channel first
+	if !empty(desired.Channel) && !empty(desired.Version) {
+		if clusterVersion.Spec.Channel != desired.Channel {
+			logger.Info(fmt.Sprintf("Setting ClusterVersion to Channel %s Version %s", desired.Channel, desired.Version))
+			clusterVersion.Spec.Channel = desired.Channel
+			err = c.client.Update(context.TODO(), clusterVersion)
+			if err != nil {
+				return false, err
+			}
+
+			// Retrieve the updated version
+			clusterVersion, err = c.GetClusterVersion()
+			if err != nil {
+				return false, err
+			}
+		}
+
+		// The CVO may need time sync the version before launching the upgrade
+		updateAvailable := false
+		for _, update := range clusterVersion.Status.AvailableUpdates {
+			if update.Version == desired.Version && update.Image != "" {
+				updateAvailable = true
+			}
+		}
+		if !updateAvailable {
+			return false, nil
 		}
 
-		// Retrieve the updated version
-		clusterVersion, err = c.GetClusterVersion()
+		clusterVersion.Spec.Overrides = []configv1.ComponentOverride{}
```

Comment:
This is to empty the contains exist in the spec.Overrides?

Reply:
Change introduced in https://github.com/openshift/managed-upgrade-operator/pull/168/files#diff-d7acacac799c3b89ebeedaaf44a111ed1cde3e3e09c3f333f67d25aa213fa96e but shown in diff as part of new block. I think the intent here is to nullify any overrides if they exist and make sure that all components are upgraded/created for the new version.